woocommerce/woocommerce-ios
# Store creation M2 - domain selection: empty/error/loading states, design updates

Parts of #8045 


### Description


After the previous basic UI integration, this PR updated the design and handled different states of the domain suggestions.

Design updates include:
- Updated the large title from a label in the view to the large title style for the navigation bar, so that it doesn't take up much vertical space on landscape and shrinks when scrolling up
- Landscape layout:
  - Previously, `List` was used for the domain results but it had its own scrolling behavior and it was hard to fit the height to its content. This PR changed it to show the domain suggestions in a `LazyVStack` (there are always max 20 results without infinite scroll) contained in a `ScrollView`. This way, the user can always scroll to the bottom of the domain list
- Search text field (`SearchHeader` component): it's currently used in `ProductSelector`and `FilterListSelector`, I tested these two use cases to make sure there are no visual changes. I added a `Customizations` struct that can be passed to the initializer, with the styles in current use cases as default
- `DomainRowView`: added padding to the row

States of domain suggestions:
- **Empty query**: whenever the query is empty (including empty spaces), a placeholder image is shown - please note that the image asset needs some updates to support dark mode HyVloP5FipZzyPVenH2euI-fi-2453%3A99694 as a separate subtask
- **Error**: an error message is shown. If it's a `DotcomError.unknown` error, the `message` string is shown (should be localized)
- **Loading**: a `ProgressView` spinner is shown
- **Results**: the domain suggestions are shown

I updated the SwiftUI previews to include previews for these 4 states for faster design updates during development. Please feel free to suggest anything, I haven't worked on SwiftUI extensively for a while.

### Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the domain selector should be shown with a placeholder image
- Enter some valid query for domain suggestions --> a spinner should be shown while loading. after a bit, some results should be shown
- Select a domain --> a checkmark should be shown on the row with bold text, and `Continue` CTA should be shown at the bottom
- Enter some invalid query (e.g. `!!!!`) --> the `Continue` CTA should be hidden, and a spinner should be shown while loading. after a bit, an error message should be shown like `Domain searches only support the following characters...`
- Feel free to test the above steps in landscape mode

### Screenshots


Please feel free to suggest any other improvements.

empty query | loading | error | results
-- | -- | -- | --
![Simulator Screen Shot - iPhone 14 - 2022-11-11 at 10 00 10](https://user-images.githubusercontent.com/1945542/201246003-ae271f13-680e-4ba4-800e-411cdde29f23.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-11 at 10 00 23](https://user-images.githubusercontent.com/1945542/201246010-f91ecc95-ed49-4ca5-b58c-7604fe948b22.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-11 at 10 00 28](https://user-images.githubusercontent.com/1945542/201246014-cbccfe55-d52d-489d-942c-6ac1e3747404.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-11 at 10 01 05](https://user-images.githubusercontent.com/1945542/201246016-20fcfd97-4c4d-43cc-966a-1324448d7793.png)


---
